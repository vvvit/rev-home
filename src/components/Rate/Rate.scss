@import '../../styles/variables.scss';

.Rate {
    position: relative;

    &-Value {
        padding: 10px 14px;
        background: $color-white;
        display: inline;
        font-size: $font-size-l;
        border: 1px solid $color-gray;
        border-radius: 20px;
        color: $color-blue;
        position: absolute;
        top: -20px;
        left: 50%;
        transform: translateX(-50%);
    }
}
