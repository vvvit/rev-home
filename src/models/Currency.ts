export const enum CurrencyName {
    USD = 'usd',
    EUR = 'eur',
    GBP = 'gbp',
}

export interface ICurrency {
    name: CurrencyName,
    value: number
}
