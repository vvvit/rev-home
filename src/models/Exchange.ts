export const enum CurrencyExchangeType {
    BUY = 'buy',
    SELL = 'sell'
}
