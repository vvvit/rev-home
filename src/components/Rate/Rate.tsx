import * as React from 'react';
import {cn} from '@bem-react/classname';

import {CurrencyName} from '../../models/Currency';

import './Rate.scss';

const cnApp = cn('Rate');

export interface RateProps {
    sellCurrency: CurrencyName | null;
    buyCurrency: CurrencyName | null;
    value: number | null;
    error: string | null;
}

export class Rate extends React.Component<RateProps> {

    render() {
        const {sellCurrency, buyCurrency, value, error} = this.props;
        if (!value && !error) {
            return null;
        }

        return (
            <div className={cnApp()}>
                <div className={cnApp('Value', {error: Boolean(error)})}>
                    {!error ? `1 ${sellCurrency} = ${value} ${buyCurrency}` : error}
                </div>
            </div>
        );
    }
}
