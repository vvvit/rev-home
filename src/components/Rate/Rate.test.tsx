import * as React from 'react';
import { shallow } from 'enzyme';

import { Rate } from './Rate';
import { CurrencyName } from '../../models/Currency';

describe('Rate', () => {
    it('renders correctly with value and without an error', () => {
        expect(shallow(
            <Rate
                sellCurrency={CurrencyName.EUR}
                buyCurrency={CurrencyName.USD}
                value={0.98}
                error={null}
            />
        )).toMatchSnapshot();
    });

    it('renders correctly with error', () => {
        expect(shallow(
            <Rate
                sellCurrency={CurrencyName.EUR}
                buyCurrency={CurrencyName.USD}
                value={0.98}
                error="api error"
            />
        )).toMatchSnapshot();
    });

    it('returns null without value and error', () => {
        const component = shallow(
            <Rate
                sellCurrency={CurrencyName.EUR}
                buyCurrency={CurrencyName.USD}
                value={null}
                error={null}
            />
        );

        expect(component.type()).toEqual(null);
    });
});
