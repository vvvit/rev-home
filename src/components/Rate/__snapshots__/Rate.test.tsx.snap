// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate renders correctly with error 1`] = `
<div
  className="Rate"
>
  <div
    className="Rate-Value Rate-Value_error"
  >
    api error
  </div>
</div>
`;

exports[`Rate renders correctly with value and without an error 1`] = `
<div
  className="Rate"
>
  <div
    className="Rate-Value"
  >
    1 EUR = 0.98 USD
  </div>
</div>
`;
